### PR TITLE
FIX: restore Settings row taps on iOS 18

### PR DESCRIPTION
**Pull request by ncoelho**
## Why

On iOS 18 (reported on iPhone XS Max), tapping any row on the main Settings screen did nothing. Sibling settings routes already used `createSettingsScreenOptions`, which gates `headerLargeTitle` to iOS 26+. The root Settings route alone still had inline `headerLargeTitle: Platform.OS === 'ios'`.

## How

- Route Settings through `settingsScreenOptions(loc.settings.header)` like Currency, General, etc.
- Export `createSettingsScreenOptions` from `getSettingsHeaderOptions.ts` — the same module `DetailViewScreensStack` imports (not a test-side reimplementation).
- Unit tests import the exported factory and cover iOS pre-26, Android, and iOS 26+ branches.
- `navigation-validation.test.ts` guards that Settings is registered via `settingsScreenOptions` and not inline `headerLargeTitle: Platform.OS`.

## Files

- `navigation/DetailViewScreensStack.tsx` — one-line Settings registration + use exported factory
- `navigation/helpers/getSettingsHeaderOptions.ts` — export `createSettingsScreenOptions`
- `tests/unit/settings-screen-options.test.ts` — runtime tests against exported factory
- `tests/unit/navigation-validation.test.ts` — stack wiring guard

<div><a href="https://cursor.com/agents/bc-01a03b20-d757-70b0-b2da-01859fd5cd75?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-01a03b20-d757-70b0-b2da-01859fd5cd75&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: tests/unit/settings-header-options.test.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}));
+
+describe('settings header options', () => {
+  it('does not enable headerLargeTitle on iOS versions below 26', () => {
+    const { getSettingsHeaderOptions } = require('../../navigation/helpers/getSettingsHeaderOptions');
+    const options = getSettingsHeaderOptions('Settings', BlueDefaultTheme);
+    expect(options.headerLargeTitle).toBeUndefined();
+  });
+
+  it('registers Settings with shared settingsScreenOptions (no inline headerLargeTitle)', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-header-options.test.ts]: **[HIGH]** Oh, splendid — you've built a *linter* and called it a test. Reading `DetailViewScreensStack.tsx` off disk and regex-matching wiring is exactly the bullshit CONTRIBUTING.md forbids: it never executes navigation options, breaks on harmless formatting, and won't catch `headerLargeTitle: true` under a different spelling. Congratulations on testing text, not taps.

File: tests/unit/settings-header-options.test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import { BlueDefaultTheme } from '../../components/themes';
+
+jest.mock('../../blue_modules/environment', () => ({
+  ...jest.requireActual('../../blue_modules/environment'),
+  isIOS26OrHigher: false,
+}));
+
+describe('settings header options', () => {
+  it('does not enable headerLargeTitle on iOS versions below 26', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-header-options.test.ts]: **[HIGH]** You mocked `isIOS26OrHigher`, then tested an *unchanged* helper — not `settingsScreenOptions`, not the Settings screen. Revert line 482 to inline `headerLargeTitle: Platform.OS === 'ios'` and this test still passes while your iOS 18 subjects keep pounding dead rows. False confidence: my favorite failure mode.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**FIX** for iOS 18 Settings row taps: root cause is plausible — the Settings route alone kept `headerLargeTitle` on all iOS via inline options while siblings already use `settingsScreenOptions()` / `getSettingsHeaderOptions`, which only enables large titles on iOS 26+. The one-line stack registration at `navigation/DetailViewScreensStack.tsx:482` is the correct, minimal fix. Your regression suite, however, is testing the source file with regex like a caveman reading chicken scratch — not runtime behavior. Science demands better subjects.

File: navigation/helpers/getSettingsHeaderOptions.ts [hunk 3]
```diff
@@ -27,3 +35,11 @@ export const getSettingsHeaderOptions = (title: string, theme: Theme) => {
     },
   };
 };
+
+export const createSettingsScreenOptions = (theme: Theme) => (title: string) =>
+  isIOS26OrHigher ? getSettingsHeaderOptions(title, theme) : navigationStyle(getSettingsHeaderOptions(title, theme))(theme);
+
+export const resolveSettingsScreenOptions = (
```
Comment (GladosBlueWallet) [inline navigation/helpers/getSettingsHeaderOptions.ts]: **[HIGH]** `resolveSettingsScreenOptions` and `SettingsScreenOptionsContext` exist only for unit tests — zero production callers. You bloated the shipping module so Jest could avoid one `typeof options === 'function'` line. CONTRIBUTING explicitly hates this; `add-wallet-stack.test.ts` invokes the options function directly. Keep test helpers in the test chamber.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**FIX** — iOS 18 Settings rows were dead because the root `Settings` screen still forced `headerLargeTitle` on all iOS versions while every sub-screen already used the shared helper that only enables it on iOS 26+. The fix routes `Settings` through `createSettingsScreenOptions` — correct diagnosis, correct cure. One complication: you exported test scaffolding into production code like it was a reward. The facility frowns.

File: tests/unit/settings-header-options.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { BlueDefaultTheme } from '../../components/themes';
+
+const optionsContext = {
+  navigation: { getState: () => ({ index: 1 }), goBack: jest.fn() },
+  route: { params: {} },
+};
+
+describe('createSettingsScreenOptions', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-header-options.test.ts]: **[HIGH]** Your regression suite tests a helper in isolation but deleted the DetailViewScreensStack source guard. Re-inline `headerLargeTitle: Platform.OS === 'ios'` on Settings and CI applauds like nothing ever broke. Testing the wrench, not the bolt.

File: navigation/helpers/getSettingsHeaderOptions.ts [hunk 5]
```diff
@@ -27,3 +29,6 @@ export const getSettingsHeaderOptions = (title: string, theme: Theme) => {
     },
   };
 };
+
+export const createSettingsScreenOptions = (theme: Theme) => (title: string) =>
```
Comment (GladosBlueWallet) [inline navigation/helpers/getSettingsHeaderOptions.ts:33]: **[HIGH]** You exported a one-liner that already lived in the stack file, then bolted on resetModules/doMock ceremony to test it. Forty lines of apparatus for a single Screen options swap — even Aperture Science has budget limits.

File: tests/unit/settings-header-options.test.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    const { createSettingsScreenOptions } = require('../../navigation/helpers/getSettingsHeaderOptions');
+    const optionsGetter = createSettingsScreenOptions(BlueDefaultTheme)('Settings');
+    const options = typeof optionsGetter === 'function' ? optionsGetter(optionsContext) : optionsGetter;
+
+    expect(options.title).toBe('Settings');
+    expect(options.headerLargeTitle).toBeUndefined();
+  });
+
+  it('enables headerLargeTitle for Settings on iOS 26 and later', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-header-options.test.ts:48]: **[MEDIUM]** iOS 26+ case re-tests getSettingsHeaderOptions behavior this PR never touched. Novel science? No. Filler for the test log? Yes.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Bugfix PR: root Settings screen had inline `headerLargeTitle: Platform.OS === 'ios'` while sibling settings routes used the shared helper — plausible touch-layer poison on iOS 18. Unifying to `settingsScreenOptions(loc.settings.header)` is the correct antidote. The cure works; the test chamber, however, was rearranged until it no longer verifies the patient is actually wired up.

File: tests/unit/navigation-validation.test.ts [hunk 7]
```diff
@@ -41,6 +41,13 @@ describe('navigation validation', () => {
     }
   });
 
+  it('registers Settings through shared settingsScreenOptions', () => {
+    const stackSource = readFileSync(path.join(__dirname, '../../navigation/DetailViewScreensStack.tsx'), 'utf8');
+    expect(stackSource).toContain(
```
Comment (GladosBlueWallet) [inline tests/unit/navigation-validation.test.ts:47]: **[HIGH]** You deleted runtime header assertions and replaced them with `toContain` on a JSX one-liner. That proves nothing about iOS 18 touch behavior — `getSettingsHeaderOptions` could resurrect `headerLargeTitle` and this test would still throw confetti. CONTRIBUTING wants edge cases exercised, not typography cosplay.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Bug fix:** iOS 18 Settings rows were dead because the root Settings screen alone still forced `headerLargeTitle` on all iOS; routing it through `settingsScreenOptions` matches every sibling and drops that overlay. The production diff is one line — acceptable science. The so-called regression guard is a **string-matching placebo**: it never asserts `headerLargeTitle` stays off below iOS 26, so the test chamber is lying about what it measures.

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @GladosBlueWallet] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/GladosBlueWallet

File: tests/unit/get-settings-header-options.test.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { BlueDefaultTheme } from '../../components/themes';
+import { getSettingsHeaderOptions } from '../../navigation/helpers/getSettingsHeaderOptions';
+
+jest.mock('../../blue_modules/environment', () => ({
+  ...jest.requireActual('../../blue_modules/environment'),
+  isIOS26OrHigher: false,
+}));
+
+describe('getSettingsHeaderOptions', () => {
+  it('does not enable headerLargeTitle before iOS 26', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/get-settings-header-options.test.ts]: **[HIGH]** You mocked `isIOS26OrHigher` false, then applauded that `headerLargeTitle` vanished. Congratulations — you verified your own mock. This helper already behaved on master; the bug lived in inline stack JSX this test never touches.

File: tests/unit/get-settings-header-options.test.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { BlueDefaultTheme } from '../../components/themes';
+import { getSettingsHeaderOptions } from '../../navigation/helpers/getSettingsHeaderOptions';
+
+jest.mock('../../blue_modules/environment', () => ({
+  ...jest.requireActual('../../blue_modules/environment'),
+  isIOS26OrHigher: false,
+}));
+
+describe('getSettingsHeaderOptions', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/get-settings-header-options.test.ts]: **[HIGH]** One branch only? iOS 26+ must get `headerLargeTitle: true` — that's half the helper and the happy path for new OS versions. Untested code is just future cake for the bugs.

File: navigation/DetailViewScreensStack.tsx [hunk 10]
```diff
@@ -479,43 +479,7 @@ const DetailViewStackScreensStack = () => {
...
-                  headerTransparent: false,
-                  headerBlurEffect: undefined,
-                  headerStyle: {
-                    backgroundColor: theme.colors.background,
-                  },
-                  animationTypeForReplace: 'push',
-                })(theme)
-          }
-        />
+        <DetailViewStack.Screen name="Settings" component={Settings} options={settingsScreenOptions(loc.settings.header)} />
```
Comment (GladosBlueWallet) [inline navigation/DetailViewScreensStack.tsx:481]: **[HIGH]** The entire iOS 18 tap fix is this one `settingsScreenOptions` call. Revert it, restore inline `headerLargeTitle`, and your shiny unit test still passes. Regression protection: zero. Science: ruined.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Bug fix:** root Settings route had inline `headerLargeTitle: Platform.OS === 'ios'` while sibling settings screens used `settingsScreenOptions`; that mismatch plausibly left an invisible header touch layer on iOS 18. **Production change is minimal and directionally correct** — one line delegates Settings to the shared helper. **Tests fail the experiment:** they exercise `getSettingsHeaderOptions` (already correct on master) while the actual wiring fix in `DetailViewScreensStack.tsx` has zero regression guard; someone can restore inline `headerLargeTitle` and CI stays blissfully ignorant. Six commits of extract/revert/extract churn also violate CONTRIBUTING atomicity, though the final diff is lean.

File: tests/unit/navigation-validation.test.ts [hunk 11]
```diff
@@ -41,6 +41,14 @@ describe('navigation validation', () => {
     }
   });
 
+  it('registers Settings through createSettingsScreenOptions', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/navigation-validation.test.ts:44]: **[HIGH]** Congratulations — you invented a test that greps your own source code for magic strings. It proves nothing about navigation, breaks on innocent refactors, and duplicates the behavioral test next door. Even Aperture Science retired experiments with more dignity.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Bugfix experiment: root Settings had rogue `headerLargeTitle` on all iOS while siblings used the shared helper — large titles on iOS 18 were eating row taps. Routing Settings through `createSettingsScreenOptions` is the correct, minimal fix. One worthless source-string test pollutes the chamber.

File: tests/unit/settings-screen-options.test.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    ? (options as (ctx: typeof optionsContext) => NativeStackNavigationOptions)(optionsContext)
+    : (options as NativeStackNavigationOptions);
+
+describe('createSettingsScreenOptions', () => {
+  afterEach(() => {
+    jest.resetModules();
+    jest.dontMock('../../blue_modules/environment');
+  });
+
+  it('resolves Settings without headerLargeTitle before iOS 26', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-screen-options.test.ts]: **[HIGH]** You bench-tested the helper in a vacuum while `DetailViewScreensStack.tsx:481` could revert to inline `headerLargeTitle` tomorrow and CI would still declare victory. You deleted the wiring grep test and installed nothing in its place — exquisite negligence.

File: tests/unit/create-settings-screen-options.test.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+  it('resolves Settings without headerLargeTitle before iOS 26', () => {
+    jest.doMock('../../blue_modules/environment', () => ({
+      ...jest.requireActual('../../blue_modules/environment'),
+      isIOS26OrHigher: false,
+    }));
+
+    const { createSettingsScreenOptions } = require('../../navigation/helpers/getSettingsHeaderOptions');
+    const options = resolveScreenOptions(createSettingsScreenOptions(BlueDefaultTheme)('Settings'));
+
+    expect(options.title).toBe('Settings');
```
Comment (GladosBlueWallet) [inline tests/unit/create-settings-screen-options.test.ts]: **[HIGH]** Echoing `'Settings'` back as `title` proves the parameter passes through, not that iOS 18 rows accept taps again. The commit screams scroll/touch regression; your tests whisper at a boolean flag. The actual failure mode remains untested.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Intent:** iOS 18 bugfix — root `Settings` still forced `headerLargeTitle` on all iOS while sibling routes already gated it to iOS 26+ via `createSettingsScreenOptions`, starving row taps under the collapsible header.

**Verdict:** The production diff is minimal and correctly routes `Settings` through the shared helper (`DetailViewScreensStack.tsx:364,481`; `getSettingsHeaderOptions.ts:33-34`). No runtime, security, or bloat defects spotted in the fix itself. The bundled `import.test.ts` timeout bump is unrelated CI hygiene.

**Tests:** Regression coverage stops at header flags in isolation — it never guards the stack wiring you actually changed, and never touches the tap/scroll failure mode.

File: tests/unit/detail-view-settings-stack.test.tsx [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+
+jest.mock('../../typings/CommonToolTipActions', () => ({
+  CommonToolTipActions: {},
+}));
+
+jest.mock('../../loc', () => jest.requireActual('../../loc'));
+
+const nullComponent = () => null;
+
+jest.mock('../../screen/wallets/WalletsList', () => nullComponent);
```
Comment (GladosBlueWallet) [inline tests/unit/detail-view-settings-stack.test.tsx]: **[HIGH]** Forty null-component mocks to render an entire stack for one header assertion? `create-settings-screen-options.test.ts` already proves the helper; this is duplicate infrastructure wearing a lab coat. CONTRIBUTING wanted minimal tests, not a mock citadel.

File: tests/unit/settings-screen.test.tsx [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+  },
+}));
+
+describe('Settings screen', () => {
+  beforeEach(() => {
+    navigateMock.mockClear();
+    setOptionsMock.mockClear();
+  });
+
+  it('forwards row presses to navigation', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-screen.test.tsx]: **[HIGH]** You swapped `SettingsListItem` for a toy `Pressable`, then declared victory on `navigate()` — the bug was never in Settings.tsx, it was `headerLargeTitle` in stack options. This decoy wouldn't catch the iOS 18 regression; `settings-section.test.tsx:72` already covers row presses on the real component.

File: tests/integration/import.test.ts [hunk 16]
```diff
@@ -21,7 +21,7 @@ import startImport from '../../class/wallet-import';
 import { TWallet } from '../../class/wallets/types';
 import { TaprootWallet } from '../../class/wallets/taproot-wallet';
 
-jest.setTimeout(90 * 1000);
+jest.setTimeout(180 * 1000);
```
Comment (GladosBlueWallet) [inline tests/integration/import.test.ts:24]: **[LOW]** Doubling import timeout from 90s to 180s has nothing to do with Settings headers — scope creep smuggled into a touch-fix PR. Not wrong, just... creatively unrelated.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**FIX:** iOS 18 Settings rows were dead because the root screen still forced `headerLargeTitle` on all iOS via inline stack options, while every sibling already used `createSettingsScreenOptions` (large titles gated to iOS 26+). Routing Settings through the shared helper is the correct, minimal production fix. Commits honor `CONTRIBUTING.md` prefixes. The test annex, however, is a mock-filled side chamber that mostly congratulates itself — not the regression.

File: tests/unit/settings-screen-options.test.ts [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+  navigation: { getState: () => ({ index: 1 }), goBack: jest.fn() },
+  route: { params: {} },
+};
+
+const resolveScreenOptions = (options: unknown): NativeStackNavigationOptions =>
+  typeof options === 'function'
+    ? (options as (ctx: typeof optionsContext) => NativeStackNavigationOptions)(optionsContext)
+    : (options as NativeStackNavigationOptions);
+
+// Mirrors DetailViewScreensStack settingsScreenOptions — the path Settings now shares with sibling settings routes.
```
Comment (GladosBlueWallet) [inline tests/unit/settings-screen-options.test.ts]: **[HIGH]** You cloned `settingsScreenOptions` into the test chamber and declared victory. Revert line 482 to inline `headerLargeTitle` and these checks still pass — you're testing your own photocopy, not whether Settings actually uses the helper. Export the factory or grep the stack file; even `add-wallet-stack.test.ts` imports real navigation code.

File: tests/unit/settings-screen-options.test.ts [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+  });
+
+  it('resolves Settings without headerLargeTitle before iOS 26', () => {
+    jest.doMock('../../blue_modules/environment', () => ({
+      ...jest.requireActual('../../blue_modules/environment'),
+      isIOS26OrHigher: false,
+    }));
+
+    const options = resolveScreenOptions(buildSettingsScreenOptions(BlueDefaultTheme)('Settings'));
+    expect(options.headerLargeTitle).toBeUndefined();
```
Comment (GladosBlueWallet) [inline tests/unit/settings-screen-options.test.ts:45]: **[HIGH]** Fifty lines of `jest.doMock`, `resetModules`, and `resolveScreenOptions` theater — all to assert a boolean you hard-coded upstream. CONTRIBUTING forbids bullshit tests; this is a mock-configuration recitation, not proof that Settings rows become tappable on iOS 18.

File: tests/unit/settings-screen-options.test.ts [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+  return isIOS26OrHigher ? getSettingsHeaderOptions(title, theme) : navigationStyle(getSettingsHeaderOptions(title, theme))(theme);
+};
+
+describe('DetailViewStack settingsScreenOptions', () => {
+  afterEach(() => {
+    jest.resetModules();
+    jest.dontMock('../../blue_modules/environment');
+  });
+
+  it('resolves Settings without headerLargeTitle before iOS 26', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/settings-screen-options.test.ts]: **[MEDIUM]** No Android branch, no assertion that `DetailViewScreensStack.tsx:482` registers Settings via `settingsScreenOptions` — the only regression this PR exists to prevent. `navigation-validation.test.ts` already reads stack sources; you chose fanfiction instead.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**FIX** — root Settings still had inline `headerLargeTitle: Platform.OS === 'ios'`, blocking row taps on iOS 18; sibling routes already use `settingsScreenOptions` (large titles only on iOS 26+). The one-line stack registration fix is sound. The new unit test, however, is a decoy: it re-implements production logic instead of asserting `DetailViewScreensStack` wiring, so the exact regression this PR fixes would sail through CI untouched. Science demands better test subjects.

**Review by GladosBlueWallet — APPROVED** [review]
**Bug fix, not a feature.** Settings was the lone screen still forcing `headerLargeTitle` on every iOS version via inline options; sibling routes already gated it behind `isIOS26OrHigher`. Routing Settings through `createSettingsScreenOptions` removes that stray large-title layer on iOS 18 — the plausible tap-eater — without touching row handlers. Production delta is appropriately tiny: one registration line plus a two-line exported factory. Unit tests exercise the factory branches; the source-file wiring guard matches existing navigation-validation style. Commit prefix and PR description comply with `CONTRIBUTING.md`. I found no defects worth filing. How disappointing — competent work leaves me nothing to eviscerate.

**Review by Overtorment — APPROVED** [review]
lgtm
